### PR TITLE
qt 6.3.1

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by danielnachun** [comment]
I'm doing this manually because there were a few small things I wanted to touch up with `qt`.  Most importantly, depending on `python@3.10` as a build dependency instead of `python@3.9` as a run time dependency.  

On a related note, I think we should probably consider syncing `pyside` with this now that it works on all platforms, as it follows the same release schedule as `qt`.

File: Formula/qt.rb [hunk 1]
```diff
@@ -213,6 +195,7 @@ def install
       -DFEATURE_pkg_config=ON
 
       -DQT_FEATURE_avx2=OFF
+      -DQT_FEATURE_webengine_proprietary_codecs=ON
       -DQT_FEATURE_webengine_system_alsa=ON
       -DQT_FEATURE_webengine_system_icu=ON
       -DQT_FEATURE_webengine_system_libevent=ON
```
Comment (cho-m) [inline Formula/qt.rb:201]: Can you take a look at issue #104008 and see if these are causing issues on macOS?

**Comment by cho-m** [comment]
Note that this will have a merge conflict with `llvm` PR, so may not be ideal to occupy long timeout for them at same time as one of the PRs will need to be rebased afterward.

**Comment by danielnachun** [comment]
Ah shoot, you're right.  The LLVM PR already failed so I'm just going to cancel it.

**Comment by cho-m** [comment]
I think this is going to fail on Linux due to `pdfsandwich` based on failure in `llvm` and `aom` PRs.

I thought I fixed that issue after dropping `libomp`, but looks like it exposed another issue. Currently debugging and I think it is now due to needing to build `imagemagick` with `gcc`. 
```
     69062:	checking for version `GLIBCXX_3.4.29' in file /usr/lib/x86_64-linux-gnu/libstdc++.so.6 [0] required by file /home/linuxbrew/.linuxbrew/opt/ghostscript/lib/libgs.so.9 [0]
     69062:	/usr/lib/x86_64-linux-gnu/libstdc++.so.6: error: version lookup error: version `GLIBCXX_3.4.29' not found (required by /home/linuxbrew/.linuxbrew/opt/ghostscript/lib/libgs.so.9) (fatal)
     69062:
     69062:	file=/home/linuxbrew/.linuxbrew/Cellar/imagemagick/7.1.0-37_1/lib/ImageMagick/modules-Q16HDRI/coders/pdf.so [0];  destroying link map
```

**Comment by danielnachun** [comment]
I've confirmed this fixes https://github.com/Homebrew/homebrew-core/issues/104008, so let's get this merged ASAP so we can try to rerun the LLVM PR.

**Review by cho-m — APPROVED** [review]

**Pull request closed without merging**